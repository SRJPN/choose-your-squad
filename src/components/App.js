import React, { useState } from "react";
import TeamModal from "./TeamModal";
import initialPlayers from "../data/players";
import Player from "./Player";

const App = () => {
  const [modalOpen, setModalOpen] = useState(false);
  const [players, setPlayers] = useState(initialPlayers);

  const onSelect = (playerId) => {
    const updatedPlayers = players.map(player => {
      if(player.id === playerId) {
        return {...player, isSelected : !player.isSelected}
      }
      return player;
    })
    setPlayers(updatedPlayers);
  }

  const selectAllPlayers = () => {
    const updatedPlayers = players.map(player => {
      return {...player, isSelected : true}
    })
    setPlayers(updatedPlayers);
  }

  const unSelectAllPlayers = () => {
    const updatedPlayers = players.map(player => {
      return {...player, isSelected : false}
    })
    setPlayers(updatedPlayers);
  }

  const selectPlayersCount = players.filter(x => x.isSelected).length;
  return (
    <div className="app">
      <header className="header">
        <div className="heading1">Squad</div>{" "}
        <div className="heading2">
          {" "}
          <span>Selected players: </span> <span>{selectPlayersCount}</span>
        </div>
        <div>
          {" "}
          <div className="button button--link">
            {" "}
            Select 10 random players{" "}
          </div>{" "}
          <div className="button button--link" onClick={selectAllPlayers}>Select all players</div>{" "}
          <div className="button button--link" onClick={unSelectAllPlayers}>Unselect all players</div>
        </div>
        <button
          onClick={() => setModalOpen(true)}
          className="button button--primary"
        >
          {" "}
          Generate teams
        </button>
      </header>

      <div className="player-list">
        {players.map((player) => <Player player={player} onSelect={() => onSelect(player.id)} key={player.id}/>)}
      </div>

      <TeamModal
        isOpen={modalOpen}
        closeModal={() => setModalOpen(false)}
        teamA={[]}
        teamB={[]}
      />
    </div>
  );
};

export default App;
