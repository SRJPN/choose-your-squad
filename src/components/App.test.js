import React from "react";
import { shallow } from "enzyme";
// import { render, screen } from '@testing-library/react';

import App from "./App";

test("should render the App correctly", () => {
  const component = shallow(<App />);
  expect(component).toMatchSnapshot();
});

test("Story 1", () => {
  const component = shallow(<App />);
  // TODO
});

test("Story 2", () => {
  const component = shallow(<App />);
  // TODO
});

test("Story 3", () => {
  const component = shallow(<App />);
  // TODO
});
