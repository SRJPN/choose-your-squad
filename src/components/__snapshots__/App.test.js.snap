// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the App correctly 1`] = `
<div
  className="app"
>
  <header
    className="header"
  >
    <div
      className="heading1"
    >
      Squad
    </div>
    <div
      className="heading2"
    >
      <span>
        Selected players: 
      </span>
      <span>
        0
      </span>
    </div>
    <div>
      <div
        className="button button--link"
      >
        Select 10 random players
      </div>
      <div
        className="button button--link"
      >
        Select all players
      </div>
      <div
        className="button button--link"
      >
        Unselect all players
      </div>
    </div>
    <button
      className="button button--primary"
      onClick={[Function]}
    >
      Generate teams
    </button>
  </header>
  <div
    className="player-list"
  >
    <div
      className="player"
      data-id="alice"
      key="alice"
    >
      <div>
        <img
          src="https://avatars.dicebear.com/api/avataaars/alice.svg?w=100"
        />
        <div>
          Alice
        </div>
      </div>
      <div
        className="player__selection"
      >
        <input
          id="alice"
          type="checkbox"
        />
        <div>
          select
        </div>
      </div>
    </div>
    <div
      className="player"
      data-id="bob"
      key="bob"
    >
      <div>
        <img
          src="https://avatars.dicebear.com/api/avataaars/bob.svg?w=100"
        />
        <div>
          Bob
        </div>
      </div>
      <div
        className="player__selection"
      >
        <input
          id="bob"
          type="checkbox"
        />
        <div>
          select
        </div>
      </div>
    </div>
    <div
      className="player"
      data-id="carol"
      key="carol"
    >
      <div>
        <img
          src="https://avatars.dicebear.com/api/avataaars/carol.svg?w=100"
        />
        <div>
          Carol
        </div>
      </div>
      <div
        className="player__selection"
      >
        <input
          id="carol"
          type="checkbox"
        />
        <div>
          select
        </div>
      </div>
    </div>
    <div
      className="player"
      data-id="dave"
      key="dave"
    >
      <div>
        <img
          src="https://avatars.dicebear.com/api/avataaars/dave.svg?w=100"
        />
        <div>
          Dave
        </div>
      </div>
      <div
        className="player__selection"
      >
        <input
          id="dave"
          type="checkbox"
        />
        <div>
          select
        </div>
      </div>
    </div>
    <div
      className="player"
      data-id="eve"
      key="eve"
    >
      <div>
        <img
          src="https://avatars.dicebear.com/api/avataaars/eve.svg?w=100"
        />
        <div>
          Eve
        </div>
      </div>
      <div
        className="player__selection"
      >
        <input
          id="eve"
          type="checkbox"
        />
        <div>
          select
        </div>
      </div>
    </div>
    <div
      className="player"
      data-id="frank"
      key="frank"
    >
      <div>
        <img
          src="https://avatars.dicebear.com/api/avataaars/frank.svg?w=100"
        />
        <div>
          Frank
        </div>
      </div>
      <div
        className="player__selection"
      >
        <input
          id="frank"
          type="checkbox"
        />
        <div>
          select
        </div>
      </div>
    </div>
    <div
      className="player"
      data-id="grace"
      key="grace"
    >
      <div>
        <img
          src="https://avatars.dicebear.com/api/avataaars/grace.svg?w=100"
        />
        <div>
          Grace
        </div>
      </div>
      <div
        className="player__selection"
      >
        <input
          id="grace"
          type="checkbox"
        />
        <div>
          select
        </div>
      </div>
    </div>
    <div
      className="player"
      data-id="heidi"
      key="heidi"
    >
      <div>
        <img
          src="https://avatars.dicebear.com/api/avataaars/heidi.svg?w=100"
        />
        <div>
          Heidi
        </div>
      </div>
      <div
        className="player__selection"
      >
        <input
          id="heidi"
          type="checkbox"
        />
        <div>
          select
        </div>
      </div>
    </div>
    <div
      className="player"
      data-id="ian"
      key="ian"
    >
      <div>
        <img
          src="https://avatars.dicebear.com/api/avataaars/ian.svg?w=100"
        />
        <div>
          Ian
        </div>
      </div>
      <div
        className="player__selection"
      >
        <input
          id="ian"
          type="checkbox"
        />
        <div>
          select
        </div>
      </div>
    </div>
    <div
      className="player"
      data-id="joan"
      key="joan"
    >
      <div>
        <img
          src="https://avatars.dicebear.com/api/avataaars/joan.svg?w=100"
        />
        <div>
          Joan
        </div>
      </div>
      <div
        className="player__selection"
      >
        <input
          id="joan"
          type="checkbox"
        />
        <div>
          select
        </div>
      </div>
    </div>
    <div
      className="player"
      data-id="karl"
      key="karl"
    >
      <div>
        <img
          src="https://avatars.dicebear.com/api/avataaars/karl.svg?w=100"
        />
        <div>
          Karl
        </div>
      </div>
      <div
        className="player__selection"
      >
        <input
          id="karl"
          type="checkbox"
        />
        <div>
          select
        </div>
      </div>
    </div>
    <div
      className="player"
      data-id="laura"
      key="laura"
    >
      <div>
        <img
          src="https://avatars.dicebear.com/api/avataaars/laura.svg?w=100"
        />
        <div>
          Laura
        </div>
      </div>
      <div
        className="player__selection"
      >
        <input
          id="laura"
          type="checkbox"
        />
        <div>
          select
        </div>
      </div>
    </div>
    <div
      className="player"
      data-id="mallet"
      key="mallet"
    >
      <div>
        <img
          src="https://avatars.dicebear.com/api/avataaars/mallet.svg?w=100"
        />
        <div>
          Mallet
        </div>
      </div>
      <div
        className="player__selection"
      >
        <input
          id="mallet"
          type="checkbox"
        />
        <div>
          select
        </div>
      </div>
    </div>
    <div
      className="player"
      data-id="naomi"
      key="naomi"
    >
      <div>
        <img
          src="https://avatars.dicebear.com/api/avataaars/naomi.svg?w=100"
        />
        <div>
          Naomi
        </div>
      </div>
      <div
        className="player__selection"
      >
        <input
          id="naomi"
          type="checkbox"
        />
        <div>
          select
        </div>
      </div>
    </div>
    <div
      className="player"
      data-id="oscar"
      key="oscar"
    >
      <div>
        <img
          src="https://avatars.dicebear.com/api/avataaars/oscar.svg?w=100"
        />
        <div>
          Oscar
        </div>
      </div>
      <div
        className="player__selection"
      >
        <input
          id="oscar"
          type="checkbox"
        />
        <div>
          select
        </div>
      </div>
    </div>
    <div
      className="player"
      data-id="peggy"
      key="peggy"
    >
      <div>
        <img
          src="https://avatars.dicebear.com/api/avataaars/peggy.svg?w=100"
        />
        <div>
          Peggy
        </div>
      </div>
      <div
        className="player__selection"
      >
        <input
          id="peggy"
          type="checkbox"
        />
        <div>
          select
        </div>
      </div>
    </div>
    <div
      className="player"
      data-id="quentin"
      key="quentin"
    >
      <div>
        <img
          src="https://avatars.dicebear.com/api/avataaars/quentin.svg?w=100"
        />
        <div>
          Quentin
        </div>
      </div>
      <div
        className="player__selection"
      >
        <input
          id="quentin"
          type="checkbox"
        />
        <div>
          select
        </div>
      </div>
    </div>
    <div
      className="player"
      data-id="rachel"
      key="rachel"
    >
      <div>
        <img
          src="https://avatars.dicebear.com/api/avataaars/rachel.svg?w=100"
        />
        <div>
          Rachel
        </div>
      </div>
      <div
        className="player__selection"
      >
        <input
          id="rachel"
          type="checkbox"
        />
        <div>
          select
        </div>
      </div>
    </div>
    <div
      className="player"
      data-id="samir"
      key="samir"
    >
      <div>
        <img
          src="https://avatars.dicebear.com/api/avataaars/samir.svg?w=100"
        />
        <div>
          Samir
        </div>
      </div>
      <div
        className="player__selection"
      >
        <input
          id="samir"
          type="checkbox"
        />
        <div>
          select
        </div>
      </div>
    </div>
    <div
      className="player"
      data-id="tara"
      key="tara"
    >
      <div>
        <img
          src="https://avatars.dicebear.com/api/avataaars/tara.svg?w=100"
        />
        <div>
          Tara
        </div>
      </div>
      <div
        className="player__selection"
      >
        <input
          id="tara"
          type="checkbox"
        />
        <div>
          select
        </div>
      </div>
    </div>
    <div
      className="player"
      data-id="ulysses"
      key="ulysses"
    >
      <div>
        <img
          src="https://avatars.dicebear.com/api/avataaars/ulysses.svg?w=100"
        />
        <div>
          Ulysses
        </div>
      </div>
      <div
        className="player__selection"
      >
        <input
          id="ulysses"
          type="checkbox"
        />
        <div>
          select
        </div>
      </div>
    </div>
    <div
      className="player"
      data-id="vera"
      key="vera"
    >
      <div>
        <img
          src="https://avatars.dicebear.com/api/avataaars/vera.svg?w=100"
        />
        <div>
          Vera
        </div>
      </div>
      <div
        className="player__selection"
      >
        <input
          id="vera"
          type="checkbox"
        />
        <div>
          select
        </div>
      </div>
    </div>
    <div
      className="player"
      data-id="walter"
      key="walter"
    >
      <div>
        <img
          src="https://avatars.dicebear.com/api/avataaars/walter.svg?w=100"
        />
        <div>
          Walter
        </div>
      </div>
      <div
        className="player__selection"
      >
        <input
          id="walter"
          type="checkbox"
        />
        <div>
          select
        </div>
      </div>
    </div>
    <div
      className="player"
      data-id="xavier"
      key="xavier"
    >
      <div>
        <img
          src="https://avatars.dicebear.com/api/avataaars/xavier.svg?w=100"
        />
        <div>
          Xavier
        </div>
      </div>
      <div
        className="player__selection"
      >
        <input
          id="xavier"
          type="checkbox"
        />
        <div>
          select
        </div>
      </div>
    </div>
    <div
      className="player"
      data-id="yvonne"
      key="yvonne"
    >
      <div>
        <img
          src="https://avatars.dicebear.com/api/avataaars/yvonne.svg?w=100"
        />
        <div>
          Yvonne
        </div>
      </div>
      <div
        className="player__selection"
      >
        <input
          id="yvonne"
          type="checkbox"
        />
        <div>
          select
        </div>
      </div>
    </div>
    <div
      className="player"
      data-id="zack"
      key="zack"
    >
      <div>
        <img
          src="https://avatars.dicebear.com/api/avataaars/zack.svg?w=100"
        />
        <div>
          Zack
        </div>
      </div>
      <div
        className="player__selection"
      >
        <input
          id="zack"
          type="checkbox"
        />
        <div>
          select
        </div>
      </div>
    </div>
  </div>
  <TeamModal
    closeModal={[Function]}
    isOpen={false}
    teamA={Array []}
    teamB={Array []}
  />
</div>
`;
